rviz_rendering/src/rviz_rendering/objects/screw_visual.cpp

```diff
@@ -0,0 +1,164 @@
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the Willow Garage, Inc. nor the names of its
+ *       contributors may be used to endorse or promote products derived from
+ *       this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+#include <OgreSceneNode.h>
+#include <OgreSceneManager.h>
+
+#include "rviz_rendering/objects/arrow.hpp"
+#include "rviz_rendering/objects/billboard_line.hpp"
+
+#include "rviz_rendering/objects/screw_visual.hpp"
+
+namespace rviz_rendering
+{
+ScrewVisual::ScrewVisual(Ogre::SceneManager * scene_manager, Ogre::SceneNode * parent_node)
+{
+  scene_manager_ = scene_manager;
+
+  // Ogre::SceneNode s form a tree, with each node storing the transform (position and orientation)
+  // of itself relative to its parent. Ogre does the math of combining those transforms
+  // for rendering. Here we create a node to store the pose of the screw's header
+  // frame relative to the RViz fixed frame.
+  frame_node_ = parent_node->createChildSceneNode();
+  linear_node_ = frame_node_->createChildSceneNode();
+  angular_node_ = frame_node_->createChildSceneNode();
+  hide_small_values_ = true;
+
+  // We create the arrow object within the frame node so that we can
+  // set its position and direction relative to its header frame.
+  arrow_linear_ = new rviz_rendering::Arrow(scene_manager_, linear_node_);
+  arrow_angular_ = new rviz_rendering::Arrow(scene_manager_, angular_node_);
+  circle_angular_ = new rviz_rendering::BillboardLine(scene_manager_, angular_node_);
+  circle_arrow_angular_ = new rviz_rendering::Arrow(scene_manager_, angular_node_);
```

Comment:
Rather than doing all of this and then having to delete it in the destructor, can we just use `std::unique_ptr` here instead?